eth/handler.go

```diff
@@ -697,7 +699,7 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		// Update the peers total difficulty if needed, schedule a download if gapped
 		if request.TD.Cmp(p.Td()) > 0 {
 			p.SetTd(request.TD)
-			td := pm.blockchain.GetTd(pm.blockchain.CurrentBlock().Hash())
+			td := pm.blockchain.GetTd(pm.blockchain.CurrentBlock().Hash(), pm.blockchain.CurrentBlock().NumberU64())
```

Comment:
This is a race condition. Current block might change between two invocations, causing td to be nil. Please query the current block once and fetch the hash/number from that.